docs/reference/snapshot-restore/restore-snapshot.asciidoc

```diff
@@ -1,187 +1,523 @@
+
+After all primary shards are recovered, the replication process creates and
+distributes replicas across eligible data nodes. When replication is complete,
+the cluster health status typically becomes `green`.
+
+You can monitor the cluster health status using the <<cluster-health,cluster
+health API>>.
+
+[source,console]
+----
+GET _cluster/health
+----
+
+To get detailed information about ongoing shard recoveries, use the
+<<indices-recovery,index recovery API>>.
+
+[source,console]
+----
+GET my-index/_recovery
+----
+// TEST[setup:setup-snapshots]
+
+To view any unassigned shards, use the <<cat-shards,cat shards API>>.
+
+[source,console]
+----
+GET _cat/shards?v=true&h=index,shard,prirep,state,node,unassigned.reason&s=state
+----
+
+Unassigned shards have a `state` of `UNASSIGNED`. The `prirep` value is `p` for
+primary shards and `r` for replicas. The `unassigned.reason` describes why the
+shard remains unassigned.
+
+To get a more in-depth explanation of an unassigned shard's allocation status,
+use the <<cluster-allocation-explain,cluster allocation explanation API>>.
+
+[source,console]
+----
+GET _cluster/allocation/explain?filter_path=index,node_allocation_decisions.node_name,node_allocation_decisions.deciders.*
+{
+  "index": "my-index",
+  "shard": 0,
+  "primary": false,
+  "current_node": "my-node"
+}
+----
+// TEST[s/^/PUT my-index\n/]
+// TEST[s/"primary": false,/"primary": false/]
+// TEST[s/"current_node": "my-node"//]
 
 [discrete]
-=== Restoring to a different cluster
+[[cancel-restore]]
+=== Cancel a restore
 
-The information stored in a snapshot is not tied to a particular cluster or a cluster name. Therefore, it's possible to
-restore a snapshot created from one cluster into another cluster by registering the repository that contains the snapshot in the new cluster and starting the restore process.
+You can delete an index or data stream to cancel its ongoing restore. This also
+deletes any existing data for the index or data stream.
 
-The topology of the new cluster does not have to match the cluster where the snapshot was created. However, the version of the new cluster must be the same or only one major version newer than the cluster that was used to create the snapshot. For example, you can restore a 5.x snapshot to a 6.x cluster, but not a 5.x snapshot to a 7.x cluster.
+[source,console]
+----
+# Delete an index
+DELETE my-index
 
-If the new cluster has a smaller capacity, consider the following changes:
+# Delete a data stream
+DELETE _data_stream/logs-my_app-default
+----
+// TEST[setup:setup-snapshots]
 
-* Ensure that the new cluster has enough capacity to store all data streams and indices in the snapshot.
-* Change the index settings during the restore operation to reduce the
-<<dynamic-index-number-of-replicas,number of replicas>>.
-* Use the `indices` parameter to choose only specific data streams or indices to restore.
+[discrete]
+[[restore-different-cluster]]
+=== Restore to a different cluster
```

Comment:
I think it would help to add that you have to register the repository with the new cluster, and mention that it should have `"readonly": true` if it's registered with another cluster too.